internal/pkg/source/source.go

```diff
@@ -382,17 +379,39 @@ func sourcesFromBlocks(fn *ssa.Function, conf classifier) []*Source {
 
 			// source obtained through a field or an index operation
 			case *ssa.Field, *ssa.FieldAddr, *ssa.IndexAddr, *ssa.Lookup:
+
+			// source map (arrays and slices have regular Allocs)
+			case *ssa.MakeMap, *ssa.MakeChan:
 			}
 
 			// all of the instructions that the switch lets through are values as per ssa/doc.go
-			if v := instr.(ssa.Value); conf.IsSource(utils.Dereference(v.Type())) {
+			if v := instr.(ssa.Value); isSourceType(conf, v.Type()) {
 				sources = append(sources, New(v.(ssa.Node), conf))
 			}
 		}
 	}
 	return sources
 }
 
+func isSourceType(c classifier, t types.Type) bool {
+	deref := utils.Dereference(t)
+	switch tt := deref.(type) {
+	case *types.Array:
+		return isSourceType(c, tt.Elem())
+	case *types.Slice:
+		return isSourceType(c, tt.Elem())
+	case *types.Chan:
+		return isSourceType(c, tt.Elem())
+	case *types.Map:
+		key := isSourceType(c, tt.Key())
+		elem := isSourceType(c, tt.Elem())
+		return key || elem
+	default:
+		u := tt.Underlying()
+		return c.IsSource(tt) || (u != tt && isSourceType(c, u))
```

Comment:
I might prefer to be explicit in which types do or don't get passed to `isSourceType`, rather than rely on the implementation detail of which are self-underlying.

```golang
	case *types.Basic, *types.Interface, *types.Tuple, *types.Struct:
		return false
	default:
		u := tt.Underlying()
		fmt.Printf("%v, underlying=%v\n", tt, u)
		return c.IsSource(tt) || isSourceType(c, u)
	}
```

At the very least, the non-obvious `u != tt` condition deserves a comment.

Reply:
Agreed, I think that reads better.

Is the `Printf` as a suggestion to make sure we weren't missing anything? (I gave it a shot and it looks good to me). Or are you suggesting that we should include that in the code?